manager/interfaces.go

```diff
@@ -82,3 +83,8 @@ type SignRequestFunc func(meta metadata.Metadata, key crypto.PrivateKey, request
 // The 'chain' and 'ca' arguments are PEM encoded and sourced directly from the
 // CertificateRequest, without any attempt to parse or decode the bytes.
 type WriteKeypairFunc func(meta metadata.Metadata, key crypto.PrivateKey, chain []byte, ca []byte) error
+
+// CreateClientFunc will return a cert-manager API client used for creating
+// objects. This is called with the metadata associated with the volume being
+// published. Useful for modifying clients to make use of CSI token requests.
+type CreateClientFunc func(meta metadata.Metadata) (cmclient.Interface, error)
```

Comment:
`ClientForMetadataFunc` instead perhaps? I think it is a small change, but sort of changes the implication from always _creating_ a client to also allow for 'returning' a client (i.e. if an implementation maintained a cache of clients).

Also - and this _isn't_ something to do here, but we should consider running codegen within this repository to avoid having this interface reference an externally generated clientset...